src/test/java/org/opensearch/neuralsearch/rest/RestNeuralStatsActionIT.java

```diff
@@ -120,6 +120,12 @@ public void test_statsFiltering() throws Exception {
         assertNotNull(getNestedValue(stats, InfoStatName.TEXT_EMBEDDING_PROCESSORS.getFullPath()));
     }
 
+    public void test_statsFiltering_invalidThrows() throws Exception {
```

Comment:
can you please change approach with testing from integ test to unit tests? We have way too many integ tests, and they are expensive. For such edge case and failure scenario unit test should do the job.

Reply:
Sure. Currently text embedding executions is the only event stat. As I add more stats I can refactor the integ tests to not use text embedding models as well so we can avoid model loading too.